static/app/utils/number/rangeMap.tsx

```diff
@@ -0,0 +1,69 @@
+import orderBy from 'lodash/orderBy';
+
+export type Range<T> = {
+  max: number;
+  min: number;
+  value: T;
+};
+
+/**
+ * Maps a set of integer ranges to the corresponding values.
+ *
+ * @example
+ * ```javascript
+ * // Map plan price to support tier
+ * const teamSizeToSupportTierMap = new RangeMap([
+ *   {min: 0, max: 10, value: 'basic'},
+ *   {min: 10, max: 30, value: 'premium'},
+ *   {min: 30, max: 50, value: 'ultra-premium'},
+ * ]);
+ *
+ * const quota = quotaToTierMap.get(0); // Tier for small teams is "basic"
+ * const quota = quotaToTierMap.get(12); // Tier for a 10-30 person team is "premium"
+ * ```
+ */
+export class RangeMap<T> {
+  ranges: Range<T>[];
+
+  constructor(ranges: Range<T>[]) {
+    // Filter out sparse array slots just in case
+    const filteredRanges = ranges.filter(Boolean);
```

Comment:
Nit, but you could do this validation on #L42 where you perform the duplicate validation as well

Reply:
That's fair, but it's nice to ensure no sparse arrays early on, so type casts like `sortedRanges[i - 1]!` definitely can't lie

Reply:
This currently also acts as an immutability guard btw. For ex if you hadn't the input via filter, and stored its reference, it would have been possible for callers to modify the ranges that are supposed to be internal. In fact, this is actually still possible as each range is an obj reference. Obv this would be unintended usage, but it could result in undefined behavior